iOS/src/toga_iOS/widgets/slider.py

```diff
@@ -11,37 +19,57 @@ class TogaSlider(UISlider):
 
     @objc_method
     def onSlide_(self, obj) -> None:
-        self.interface.on_change(None)
+        self.impl.on_change()
 
+    @objc_method
+    def onPress_(self, obj) -> None:
+        self.impl.on_press()
+
+    @objc_method
+    def onRelease_(self, obj) -> None:
+        self.impl.on_release()
 
-class Slider(Widget):
+
+class Slider(Widget, toga.widgets.slider.ContinuousSliderImpl):
     def create(self):
         self.native = TogaSlider.alloc().init()
         self.native.interface = self.interface
         self.native.impl = self
 
-        self.native.continuous = True
-        self.native.addTarget(
-            self.native,
-            action=SEL("onSlide:"),
-            forControlEvents=UIControlEventValueChanged,
-        )
+        for method, events in [
+            ("onSlide", UIControlEventValueChanged),
+            ("onPress", UIControlEventTouchDown),
+            (
+                "onRelease",
+                UIControlEventTouchUpInside
+                | UIControlEventTouchUpOutside
+                | UIControlEventTouchCancel,
+            ),
+        ]:
+            self.native.addTarget(
+                self.native, action=SEL(f"{method}:"), forControlEvents=events
+            )
```

Comment:
Functionally all correct; but I'm not sure abstracting this as a loop actually adds that much clarity. Three explicit addTarget calls doesn't require the mental mapping of method and events into the call.

Reply:
Done.